base/associative.jl

```diff
@@ -451,6 +451,16 @@ function rehash!(t::ObjectIdDict, newsz = length(t.ht))
     t
 end
 
+function sizehint!(t::ObjectIdDict, newsz)
+    newsz = _tablesz(newsz*2)  # *2 for keys and values in same array
+    oldsz = length(t.ht)
+    # grow at least 25%
+    if newsz < (oldsz*5)>>2
+        return t
```

Comment:
The comment and conditional seemed to contradict each other on first read. That is, IIUC the conditional causes no growth if `newsz` is less than 25% larger than `oldsz`, whereas the comment seems to suggest that if `newsz` is less than 25% larger than `oldsz`, growth by at least 25% should occur. IIUC, perhaps worth revising the comment slightly? Something like "# do not grow if newsz is less than 25% larger than oldsz"?